src/emulator/storage/files.ts

```diff
@@ -125,12 +156,72 @@ export class StorageLayer {
     private _cloudFunctions: StorageCloudFunctions
   ) {}
 
+  /**
+   * Create SignedUrl
+   * @date 6/28/2023 - 2:56:09 PM
+   *
+   * @async
+   * @param {GetObjectRequest} request
+   * @returns {Promise<boolean>}
+   */
+  async generateSignedUrl(request: CreateSignedUrl): Promise<SignedUrlResponse> {
+    const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
+
+    const currentDate = this.getCurrentDate();
+
+    const timeToLive = request.ttlSeconds;
+    if (!Number.isInteger(timeToLive)) {
+      throw new BadRequestError("Invalid TTL Parameter");
+    }
+
+    if (timeToLive < SIGNED_URL_MIN_TTL_SECONDS || timeToLive > SIGNED_URL_MAX_TTL_SECONDS) {
+      throw new BadRequestError(`TTL specified is less than 0 or more than allowed max (2 week)`);
+    }
+
+    const authorized = await this._rulesValidator.validate(
+      ["b", request.bucketId, "o", request.decodedObjectId].join("/"),
+      request.bucketId,
+      RulesetOperationMethod.GET,
+      { before: metadata?.asRulesResource() },
+      this._projectId,
+      request.authorization
+    );
+
+    if (!authorized) {
+      throw new ForbiddenError();
+    }
+
+    if (!metadata) {
+      throw new NotFoundError("Object in bucket does not exist");
+    }
+
+    const unsignedUrl = createUnsignedUrl({
+      bucketId: request.bucketId,
+      decodedObjectId: request.decodedObjectId,
+      url: request.originalUrl,
+      urlUsableSeconds: currentDate,
+      urlTtlSeconds: request.ttlSeconds,
+    });
+
+    const signature = createSignature(unsignedUrl);
+
+    const signedUrl = `${unsignedUrl}&X-Firebase-Signature=${encodeURIComponent(signature)}`;
+
+    return {
+      signed_url: signedUrl,
+    };
+  }
+
   createBucket(id: string): void {
     if (!this._buckets.has(id)) {
       this._buckets.set(id, new CloudStorageBucketMetadata(id));
     }
   }
 
+  getCurrentDate() {
```

Comment:
Add a comment above this with the intended resulting timestamp format: `YYYYMMDD'T'HHMMSS'Z'`